core/src/main/java/jenkins/security/ResourceDomainRecommendation.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2019 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security;
+
+import hudson.Extension;
+import hudson.model.AdministrativeMonitor;
+import hudson.model.DirectoryBrowserSupport;
+import hudson.util.HttpResponses;
+import jenkins.util.SystemProperties;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import java.io.IOException;
+
+/**
+ * Recommend use of {@link ResourceDomainConfiguration} to users with the system property
+ * <code>hudson.model.DirectoryBrowserSupport.CSP</code> set to override
+ * {@link DirectoryBrowserSupport#DEFAULT_CSP_VALUE}.
+ *
+ * @see ResourceDomainConfiguration
+ *
+ * @since TODO
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class ResourceDomainRecommendation extends AdministrativeMonitor {
+
+    @Override
+    public String getDisplayName() {
+        return Messages.ResourceDomainConfiguration_DisplayName();
+    }
+
+    @Override
+    public boolean isActivated() {
+        boolean isResourceRootUrlSet = ResourceDomainConfiguration.isResourceDomainConfigured();
+        boolean isOverriddenCSP = SystemProperties.getString(DirectoryBrowserSupport.class.getName() + ".CSP") != null;
+        return isOverriddenCSP && !isResourceRootUrlSet;
+    }
+
+    @RequirePOST
+    public HttpResponse doAct(@QueryParameter String redirect, @QueryParameter String dismiss) throws IOException {
+        if (dismiss != null) {
+            disable(true);
+            return HttpResponses.redirectViaContextPath("manage");
+        }
+        return HttpResponses.redirectViaContextPath("configureSecurity");
```

Comment:
Still redirects to `/configureSecurity/` when the option is now in `/configure`.